util/encoding.go

```diff
@@ -0,0 +1,31 @@
+package util
+
+import (
+	"bytes"
+	"encoding/binary"
+)
+
+// Helpers functions regarding the encoding of basic types ( integers ...)
```

Comment:
Indeed i've tried with uint => UInt64.
The reason i've fallen back to Int is for "general usage" & compatibility. For example, in promise.go, you must add a Response at a specific index i which is an int. I could write another method with UInt, but then i'd had to do a lot of cast to operate with the rest of the code. 
I agree that using uint is much better and that was my first approach, but since the rest of the library only uses int, it becomes more of a burden to switch to uint that to stay with int.
What do you think ?